### PR TITLE
Validation cleanup

### PR DESCRIPTION
**Pull request by nmonterroso**
@drsnyder @ArturKlajnerok 
https://wikia-inc.atlassian.net/browse/SERVICES-355

Adding the validators for username/email existing after the changes in https://github.com/Wikia/helios/pull/50. Also removing some deprecated code.

### CODE REVIEW

File: controllers/users_controller.go [hunk 1]
```diff
@@ -105,19 +111,17 @@ func (controller *UsersController) usersHandler(w http.ResponseWriter, r *http.R
...
-		modelUser := models.FromDomain(user)
-
-		if controller.createUser(resp, modelUser) {
-			controller.onUserCreationComplete(modelUser, controller.getLangCodeFromRequest(r))
-		}
 	}
+
+	modelUser := models.FromDomain(user)
+	controller.onUserCreationComplete(modelUser, controller.getLangCodeFromRequest(r))
+	controller.setOutputSuccess(resp, "User created")
```
Comment (drsnyder) [inline controllers/users_controller.go]: Should this be in an `else` block? Also, I think we are missing the case where `! userValidationErrors.IsEmpty()`.

File: models/user_storage.go [hunk 2]
```diff
@@ -109,55 +109,12 @@ func (userStorage *UserStorage) LookupByEmail(email string) (*User, error) {
...
-		return false, err
-	}
-
-	return count > 0, nil
-}
-
-// Deprecated
 func (userStorage *UserStorage) FindByName(userName string, mustExist bool) (*User, error) {
 	user := new(User)
 	query := fmt.Sprintf("select * from %s where user_name=?", userStorage.tableName)
```
Comment (drsnyder) [inline models/user_storage.go]: I think this one can go away too.

**Comment by drsnyder** [comment]
Net negative!

File: controllers/users_controller.go (hunk 1 above)
Comment (nmonterroso) [inline controllers/users_controller.go]: Yah, the tests noticed this for me, and i fixed it in 400e6f5.

File: models/user_storage.go (hunk 2 above)
Comment (nmonterroso) [inline models/user_storage.go]: I wish. It's being used in [oauth_controller](https://github.com/Wikia/helios/blob/master/controllers/oauth_controller.go#L98)

File: service/user.go [hunk 3]
```diff
@@ -45,7 +46,7 @@ func (service *UserServiceValue) CreateUser(user domain.User) (bool, *errorlist.
 
 	existingEmail, err := service.userDAO.LookupByEmail(user.Email())
 	if err != nil {
-		return false, nil
+		return false, validationErrors
```
Comment (drsnyder) [inline service/user.go]: :+1: better than `nil`

File: controllers/users_controller.go [hunk 4]
```diff
@@ -105,18 +111,16 @@ func (controller *UsersController) usersHandler(w http.ResponseWriter, r *http.R
 	}
 
 	user := createDomainUserFromForm(r)
-	isDomainObjectValid, userValidationErrors := controller.userValidator.Validate(user)
+	created, userValidationErrors := controller.userService.CreateUser(user)
 
-	if userValidationErrors.ContainsError(validation.GenericDataAccessError) {
+	if created {
+		modelUser := models.FromDomain(user)
+		controller.onUserCreationComplete(modelUser, controller.getLangCodeFromRequest(r))
```
Comment (drsnyder) [inline controllers/users_controller.go]: In hindsight, we should have moved this to the service.

**Pull request merged**